src/app/core/status/status.service.ts

```diff
@@ -0,0 +1,144 @@
+import { Injectable, Injector } from '@angular/core';
+import { Subscription } from 'rxjs/Subscription';
+import { Subject } from 'rxjs/Subject';
+
+import { ModalsService } from '../../modals/modals.service';
+import { RPCService } from '../../core/rpc/rpc.service';
+import { PeerService } from '../../core/rpc/peer.service';
+
+@Injectable()
+export class StatusService {
+
+  private _subBlockInternal: Subscription;
+  private _subBlockNetwork: Subscription;
+  private _modalsService: ModalsService;
+
+  private highestBlockHeightNetwork: number = -1;
+  private highestBlockHeightInternal: number = -1;
+  private startingBlockCount: number = -1;
+  private totalRemainder: number = -1;
+
+  public statusUpdates: Subject<any> = new Subject<any>();
+  private status: any = {
+    syncPercentage: 0,
+    remainingBlocks: undefined,
+    lastBlockTime: undefined,
+    increasePerMinute: 0,
+    estimatedTimeLeft: undefined,
+    manuallyOpened: false
+  };
+
+  constructor(
+    private _peerService: PeerService,
+    private _rpcService: RPCService,
+    private _injector: Injector
+  ) {
+    this._subBlockInternal = this._peerService.getBlockCount()
+      .subscribe(
+        height => {
+          // TODO lastBlockTime
+          const lastBlockTime = new Date();
+          this.calculateSyncingDetails(lastBlockTime, height);
+          this.highestBlockHeightInternal = height;
+          this.status.lastBlockTime = lastBlockTime;
+          if (this.startingBlockCount === -1) {
+            this.startingBlockCount = height;
+          }
+        },
+        error => console.log('SyncingComponent subscription error:' + error));
+    this._subBlockNetwork = this._peerService.getBlockCountNetwork()
+      .subscribe(
+        height => {
+          this.highestBlockHeightNetwork = height;
+          if (this.totalRemainder === -1 && this.startingBlockCount !== -1) {
+            this.totalRemainder = height - this.startingBlockCount;
+          }
+        },
+        error => console.log('SyncingComponent subscription error:' + error));
+    this._rpcService.poll();
+  }
+
+  setManuallyOpened() {
```

Comment:
I've ran into the same issue with the modal service, I can't open a modal and pass extra arguments to it..
Storing the state of the syncing modal in the status service doesn't feel like a really good solution to me :P